lib/AST/ConformanceLookup.cpp

```diff
@@ -916,3 +916,11 @@ bool TypeBase::isEscapable() {
   auto canType = preprocessTypeForInvertibleQuery(this);
   return conformsToInvertible(canType, InvertibleProtocolKind::Escapable);
 }
+
+bool TypeBase::isEscapable(GenericSignature sig) {
```

Comment:
Is this new utility method really necessary? If you know `type` is an interface type, this will work whether sig is null or not:

```
GenericEnvironment::mapTypeIntoContext(sig.getGenericEnvironment(), type)->isEscapable()
```